lib/sales_analyst.rb

```diff
@@ -80,11 +86,129 @@ def standard_deviation(values, mean)
     std_dev.round(2)
   end
 
+  def average_invoices_per_merchant
+    average = all_invoices.length.to_f / all_merchants.length
+    average.round(2)
+  end
+
+  def num_of_invoices_per_merchant
+    all_merchants.each_with_object({}) do |merchant, total_per_merchant|
+      total_invoices = all_invoices.count do |invoice|
+        invoice.merchant_id == merchant.id
+      end
+      total_per_merchant[merchant] = total_invoices
+    end
+  end
+
+  def average_invoices_per_merchant_standard_deviation
+    mean = average_invoices_per_merchant
+    invoice_counts = num_of_invoices_per_merchant.values
+    std_dev = standard_deviation(invoice_counts, mean)
+  end
+
+  def top_merchants_by_invoice_count
+    mean = average_invoices_per_merchant
+    std_dev = average_invoices_per_merchant_standard_deviation
+
+    z = standard_deviations_of_mean(mean, std_dev, 2)
```

Comment:
I remember in one of the lessons they advised us to steer away from using letters as variable names unless it relates directly to the material. Just a thought for refactoring! ✅

Reply:
Good catch!